crates/wassette/src/lib.rs

```diff
@@ -198,577 +300,241 @@ pub struct ComponentInstance {
-            if policy_path.exists() {
-                match tokio::fs::read_to_string(&policy_path).await {
-                    Ok(policy_content) => match PolicyParser::parse_str(&policy_content) {
-                        Ok(policy) => {
-                            match wasistate::create_wasi_state_template_from_policy(
-                                &policy,
-                                plugin_dir.as_ref(),
-                                &environment_vars,
-                                None, // No secrets during initial loading
-                            ) {
-                                Ok(wasi_template) => {
-                                    policy_registry
-                                        .component_policies
-                                        .insert(name.clone(), Arc::new(wasi_template));
-                                    info!(component_id = %name, "Restored policy association from co-located file");
-                                }
-                                Err(e) => {
-                                    warn!(component_id = %name, error = %e, "Failed to create WASI template from policy");
-                                }
-                            }
-                        }
-                        Err(e) => {
-                            warn!(component_id = %name, error = %e, "Failed to parse co-located policy file");
-                        }
-                    },
-                    Err(e) => {
-                        warn!(component_id = %name, error = %e, "Failed to read co-located policy file");
-                    }
-                }
+        for component_id in registered_ids {
+            if let Err(error) = self.restore_policy_attachment(&component_id).await {
+                warn!(component_id = %component_id, %error, "Failed to restore policy attachment");
             }
         }
 
-        // Make sure the plugin dir exists and also create a subdirectory for temporary staging of downloaded files
-        tokio::fs::create_dir_all(&plugin_dir)
-            .await
-            .context("Failed to create plugin directory")?;
-        tokio::fs::create_dir_all(plugin_dir.as_ref().join(DOWNLOADS_DIR))
-            .await
-            .context("Failed to create downloads directory")?;
-
-        info!("LifecycleManager initialized successfully");
-        Ok(Self {
-            engine,
-            linker,
-            components: Arc::new(RwLock::new(components)),
-            registry: Arc::new(RwLock::new(registry)),
-            policy_registry: Arc::new(RwLock::new(policy_registry)),
-            oci_client: Arc::new(oci_wasm::WasmClient::new(oci_client)),
-            http_client,
-            plugin_dir: plugin_dir.as_ref().to_path_buf(),
-            environment_vars,
-            secrets_manager,
-        })
+        info!("LifecycleManager eagerly loaded components");
```

Comment:
Nit: I don't think "eagerly" adds anything here. This should probably just say "LifecycleManager finished loading components"